Tests/Pcap++Test/Tests/LiveDeviceTests.cpp

```diff
@@ -379,91 +380,192 @@ PTF_TEST_CASE(TestPcapLiveDeviceStatsMode)
+	SystemCommandTeardown iptablesDeleteOutputDrop("sudo iptables -D OUTPUT -o " + interfaceName + " -j DROP");
+
+	// open device
+	pcpp::PcapLiveDevice::DeviceConfiguration newConfig;
+	newConfig.usePoll = true;
+
+	PTF_ASSERT_TRUE(liveDev->open(newConfig));
+	DeviceTeardown devTeardown(liveDev);
+
+	int packetCount = 0;
+	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeTimeout, &packetCount, 2), -1);
+	PTF_ASSERT_EQUAL(packetCount, 0);
 
 	liveDev->close();
-	PTF_ASSERT_FALSE(liveDev->isOpened());
+#else
+	PTF_SKIP_TEST("This test can not run in Windows environment");
+#endif
+} // TestPcapLiveDeviceBlockingMode
 
-	// a negative test
-	pcpp::Logger::getInstance().suppressLogs();
-	PTF_ASSERT_FALSE(liveDev->startCapture(packetArrives, &packetCount));
-	pcpp::Logger::getInstance().enableLogs();
+PTF_TEST_CASE(TestPcapLiveDeviceBlockingModeNotTimeoutWithoutPoll)
+{
+#if !defined(_WIN32)
+	pcpp::PcapLiveDevice* liveDev = pcpp::PcapLiveDeviceList::getInstance().getPcapLiveDeviceByIp(PcapTestGlobalArgs.ipToSendReceivePackets.c_str());
+	auto interfaceName = liveDev->getName();
+
+	// drop all packets on the interface
+	auto iptablesAddInputDrop = "sudo iptables -A INPUT -i " + interfaceName + " -j DROP";
+	auto iptablesAddOutputDrop = "sudo iptables -A OUTPUT -o " + interfaceName + " -j DROP";
+	PTF_ASSERT_GREATER_OR_EQUAL_THAN(std::system(iptablesAddInputDrop.c_str()), 0);
+	PTF_ASSERT_GREATER_OR_EQUAL_THAN(std::system(iptablesAddOutputDrop.c_str()), 0);
+
+	// recover the interface at the end
+	SystemCommandTeardown iptablesDeleteInputDrop("sudo iptables -D INPUT -i " + interfaceName + " -j DROP");
+	SystemCommandTeardown iptablesDeleteOutputDrop("sudo iptables -D OUTPUT -o " + interfaceName + " -j DROP");
+
+	// open device
+	pcpp::PcapLiveDevice::DeviceConfiguration newConfig;
+	newConfig.usePoll = false; // explicitly mention not using poll
+
+	PTF_ASSERT_TRUE(liveDev->open(newConfig));
+	DeviceTeardown devTeardown(liveDev);
+
+	int packetCount = 0;
+
+	auto func = [&]() -> int
+	{
+		return liveDev->startCaptureBlockingMode(packetArrivesBlockingModeTimeout, &packetCount, 2); // ideally, it should timeout after 2 seconds
+	};
+
+	// test it not timeout after 2 seconds
+	std::packaged_task<int()> task(func);
+
+    std::future<int> future = task.get_future();
+
+    // Start a new thread to execute `startCaptureBlockingMode` and to test it timeout
+    std::thread thread(std::move(task));
+
+    // Wait for the function to finish or the timeout to occur
+    auto status = future.wait_for(std::chrono::milliseconds(3000)); // already wait for 3 seconds, and we know `startCaptureBlockingMode` is still blocking
+
+	// the function doesn't timeout
+	PTF_ASSERT_TRUE(status == std::future_status::timeout);
+
+	// restore the interface to let the callback receive packets so the thread can be jointed.
+	iptablesDeleteInputDrop.~SystemCommandTeardown();
+	iptablesDeleteOutputDrop.~SystemCommandTeardown();
```

Comment:
Calling the teardown d'tor explicitly kind of beats the purpose of the teardown class 🤔

Reply:
True, let me call `std::system` directly here